### PR TITLE
Update edit-save documentation 

### LINKED ISSUE #4070: Documentation: More examples of save() callback of block
After spending some time teaching myself Gutenberg, I found the information on the save() callback for blocks and how serializing data to comments works
https://wordpress.org/gutenberg/handbook/block-edit-save/

* More examples of save() function, showing how you would use it for
  * Serializing attributes to main comment
  * Storing attributes in child elements
  * Saving content to be rendered in the front-end using same React component as used to render preview in edit callback.
 
In regards to serialization of data, it would be helpful if it was explained how automatic serialization of attributes work. A link to the section on attributes is also needed.

### PR DESCRIPTION
**Pull request by mkaz**
## Description

* Add ES5 code samples for `edit()` documentation
* Add links to additional information around attributes
* Update save documentation with additional examples

[View branch documentation](https://github.com/WordPress/gutenberg/blob/docs/4070/save/docs/designers-developers/developers/block-api/block-edit-save.md).

Fixes #4070

## How has this been tested?

Confirm example code works as explained.

## Types of changes

Documentation.

### CODE REVIEW

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 1]
```diff
@@ -6,63 +6,145 @@ When registering a block, the `edit` and `save` functions provide the interface
...
+		'div',
+		null,
+		'Your block.'
+	);
 }
 ```
+{% ESNext %}
+```jsx
+edit() {
+	return <div> Your block. </div>;
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/block-api/block-edit-save.md]: Maybe worth removing the spaces here?

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 2]
```diff
@@ -146,6 +232,130 @@ save( { attributes } ) {
 ```
 {% end %}
 
+
+When saving your block, you want save the attributes in the same format specified by the attribute source definition.  If no attribute source is specified, the attribute will be saved to the block's comment delimiter. See the [Block Attributes documentation](/docs/designers-developers/developers/block-api/block-attributes.md) for more details.
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/block-api/block-edit-save.md]:
```suggestion
When saving your block, you want to save the attributes in the same format specified by the attribute source definition. If no attribute source is specified, the attribute will be saved to the block's comment delimiter. See the [Block Attributes documentation](/docs/designers-developers/developers/block-api/block-attributes.md) for more details.
```

**Review by chrisvanpatten — DISMISSED** [review]
Might be worthwhile to get more eyes on this, but overall it seemed good to me!

File: docs/designers-developers/developers/block-api/block-edit-save.md (hunk 1 above)
Comment (mkaz) [reply docs/designers-developers/developers/block-api/block-edit-save.md]: Removed.

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 3]
```diff
@@ -6,63 +6,145 @@ When registering a block, the `edit` and `save` functions provide the interface
...
-// Defining the edit interface
 edit( { attributes, className } ) {
 	return <div className={ className }>{ attributes.content }</div>;
 }
 ```
+{% end %}
 
 ### isSelected
 
 The isSelected property is an object that communicates whether the block is currently selected.
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]: I thought `isSelected` was a boolean?

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 4]
```diff
@@ -79,6 +161,7 @@ edit( { attributes, setAttributes, className, isSelected } ) {
 	);
 }
 ```
+{% end %}
 
 When using attributes that are objects or arrays it's a good idea to copy or clone the attribute prior to updating it:
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]: I'd love if we could ES5y the following example as well :)

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 5]
```diff
@@ -146,6 +232,130 @@ save( { attributes } ) {
...
+attributes: {
+	content: {
+		type: 'string',
+		source: 'html',
+		selector: 'p'
+	}
+},
+
+edit: ( props ) => {
+	updateFieldValue = function( val ) {
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]:
```suggestion
	var updateFieldValue = function( val ) {
```

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 6]
```diff
@@ -146,6 +232,130 @@ save( { attributes } ) {
...
+attributes: {
+	content: {
+		type: 'string',
+		source: 'html',
+		selector: 'p'
+	}
+},
+
+edit: ( { attributes, setAttributes } ) => {
+	updateFieldValue = ( val ) {
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]:
```suggestion
	const updateFieldValue = ( val ) {
```

Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]:
```suggestion
	const updateFieldValue = ( val ) => {
```

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 7]
```diff
@@ -6,63 +6,145 @@ When registering a block, the `edit` and `save` functions provide the interface
...
 
 ### setAttributes
 
 This function allows the block to update individual attributes based on user interactions.
 
+{% codetabs %}
+{% ES5 %}
 ```js
-// Defining the edit interface
+edit: ( props ) => {
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]:
```suggestion
edit( props ) {
```

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 8]
```diff
@@ -6,63 +6,145 @@ When registering a block, the `edit` and `save` functions provide the interface
...
+		[
+			count,
+			props.isSelected ? wp.element.createElement(
+				'button',
+				{ onClick: toggleSetting },
+				'Toggle setting'
+			) : null
+		]
+	);
+},
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]:
```suggestion
}
```

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 9]
```diff
@@ -146,6 +232,130 @@ save( { attributes } ) {
...
+```js
+attributes: {
+	content: {
+		type: 'string',
+		source: 'html',
+		selector: 'p'
+	}
+},
+
+edit: ( props ) => {
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]:
```suggestion
edit( props ) {
```

**Comment by chrisvanpatten** [comment]
(Dismissed my own review; if someone else approves I don't want my stale review to block a merge!)

File: docs/designers-developers/developers/block-api/block-edit-save.md (hunk 3 above)
Comment (mkaz) [reply docs/designers-developers/developers/block-api/block-edit-save.md]: Isn't a boolean an object too :-)

File: docs/designers-developers/developers/block-api/block-edit-save.md (hunk 4 above)
Comment (mkaz) [reply docs/designers-developers/developers/block-api/block-edit-save.md]: Added a clone example in ES5, so much more verbose :-)

File: docs/designers-developers/developers/block-api/block-edit-save.md (hunk 3 above)
Comment (nosolosw) [reply docs/designers-developers/developers/block-api/block-edit-save.md]: It's `0`s and `1`s as well! The more specific, the better.

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 10]
```diff
@@ -6,64 +6,146 @@ When registering a block, the `edit` and `save` functions provide the interface
...
+	let content = props.attributes.content;
+	let mySetting = props.attributes.mySetting;
+
+	// Toggle a setting when the user clicks the button
+	let toggleSetting = () => props.setAttributes( { mySetting: ! mySetting } );
+	return wp.element.createElement(
+		'div',
+		{ className: props.className },
+		[
+			count,
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]:
```suggestion
			content,
```

File: docs/designers-developers/developers/block-api/block-edit-save.md [hunk 11]
```diff
@@ -79,22 +161,45 @@ edit( { attributes, setAttributes, className, isSelected } ) {
...
 ```
+{% end %}
 
 When using attributes that are objects or arrays it's a good idea to copy or clone the attribute prior to updating it:
 
+{% codetabs %}
+{% ES5 %}
 ```js
-// Good - here a new array is created from the old list attribute and a new list item:
+// Good - cloning the old list to a new array and then adding item
```
Comment (nosolosw) [inline docs/designers-developers/developers/block-api/block-edit-save.md]: We can simplify this a bit:

 ```js
// Good - cloning the old list
var newList = attributes.list.slice();

var addListItem = function( newListItem ) {
	setAttributes( { list: newList.concat( [ newListItem ] ) } );
};

// Bad - the list from the existing attribute is modified directly to add the new list item:
var list = attributes.list;

var addListItem = function( newListItem ) {
	list.push( newListItem );
	setAttributes( { list: list } );
};
```

**Review by oandregal — APPROVED** [review]
Left a couple of comments that I'd like us to address. Other than that this is good to go.

This improves things a lot! :star2:

File: docs/designers-developers/developers/block-api/block-edit-save.md (hunk 11 above)
Comment (mkaz) [reply docs/designers-developers/developers/block-api/block-edit-save.md]: :+1: Thanks!

**Pull request merged**